src/coordinator/tsdb/remote/compressed_codecs.go

```diff
@@ -252,16 +327,22 @@ func blockReadersFromCompressedSegments(
 }
 
 /*
-SeriesIteratorFromCompressedSeries creates a SeriesIterator from a compressed protobuf
-This is the reverse of CompressedSeriesFromSeriesIterator, and takes an optional iteratorPool
+Creates a SeriesIterator from a compressed protobuf. This is the reverse of
+CompressedSeriesFromSeriesIterator, and takes an optional iteratorPool
 argument that allows reuse of the underlying iterator pools from the m3db session
 */
-func SeriesIteratorFromCompressedSeries(
+func seriesIteratorFromCompressedSeries(
 	timeSeries *rpc.Series,
 	iteratorPools encoding.IteratorPools,
-) encoding.SeriesIterator {
+) (encoding.SeriesIterator, error) {
 	initialize.Do(initializeVars)
 
+	// Attempt to decompress compressed tags first as this is the only scenario that is expected to fail
+	tagIter, err := tagIteratorFromSeries(timeSeries, iteratorPools)
```

Comment:
tagIteratorFromSeries can either decompress tags or create them using series.GetTags(). Should we only allow the first method ? This will enforce that if the series is compressed, then its tags will also always be compressed.

Reply:
Not necessarily, since series can be compressed but not have compressed tags if no iterator pools are passed in.